Packs/Hackuity/Integrations/Hackuity/Hackuity.py

```diff
@@ -0,0 +1,694 @@
+    return None
+
+
+def none_or_apply(text: Optional[str], apply: Callable[[str], Any]) -> Optional[Any]:
+    if text is None:
+        return None
+    return apply(text)
+
+
+""" COMMAND FUNCTIONS """
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication"""
+    # excption formating in case of authentication error
+    # is already done in client.get_access_token
+    set_integration_context({})  # reset cache
+    client.fetch_user()
+    return "ok"
+
+
+def hackuity_search_findings_command(
+    client: Client, args: Dict[str, Any]
+) -> CommandResults:
+    asset_name = args.get("asset_name")
+    asset_type = args.get("asset_type")
+    attribute = args.get("attribute")
+    limit = int(args["limit"])
+    cvss_min = none_or_apply(args.get("cvss_min"), float)
+    cvss_max = none_or_apply(args.get("cvss_max"), float)
+    vuln_type = args.get("vuln_type")
+    trs_min = none_or_apply(args.get("trs_min"), int)
+    trs_max = none_or_apply(args.get("trs_max"), int)
+    findings = client.fetch_findings(
+        asset_name=asset_name,
+        asset_type=asset_type,
+        attribute=attribute,
+        cvss_min=cvss_min,
+        cvss_max=cvss_max,
+        limit=limit,
+        trs_min=trs_min,
+        trs_max=trs_max,
+        vuln_type=vuln_type,
+    )
+    outputs = []
+    for finding in findings:
+        output = json_format(
+            finding,
+            {
+                "asset_id": ["Asset", "ID"],
+                "asset_name": ["Asset", "Name"],
+                "asset_type": ["Asset", "Type"],
+                "attributes": ["Attributes"],
+                "cvss": ["Score", "CVSS"],
+                "id": ["ID"],
+                "ignored": ["Status", "Ignored"],
+                "state": ["Status", "State"],
+                "sub_state": ["Status", "SubState"],
+                "trs": ["Score", "TRS"],
+                "vuln_type": ["VulnType", "ID"],
+            },
+        )
+        output["VulnType"]["Name"] = client.get_vulnerability_name(finding["vuln_type"])
+        outputs.append(output)
+    return CommandResults(
+        outputs_prefix="Hackuity.Findings",
+        outputs=outputs,
+        readable_output=tableToMarkdown(
+            "Findings",
+            outputs,
+            headers=["Asset", "VulnType", "Attributes", "Score", "Status"],
+        ),
+    )
+
+
+def hackuity_search_vulnerabilities_command(
+    client: Client, args: Dict[str, Any], hy_global_only: bool
+) -> CommandResults:
+    asset_name = args.get("asset_name")
+    asset_type = args.get("asset_type")
+    attribute = args.get("attribute")
+    limit = int(args["limit"])
+    cvss_min = none_or_apply(args.get("cvss_min"), float)
+    cvss_max = none_or_apply(args.get("cvss_max"), float)
+    vuln_type = args.get("vuln_type")
+    trs_min = none_or_apply(args.get("trs_min"), int)
+    trs_max = none_or_apply(args.get("trs_max"), int)
+    aggfindings = client.fetch_aggfindings(
```

Comment:
make sure in case there's non, a readout output is returned and not an error

Reply:
If there are no aggfindings, `fetch_aggfindings` will return an empty list, I think this is ok

![screenshot-2022-01-26-18:19:02](https://user-images.githubusercontent.com/3994389/151213549-5da56bab-f9d1-4b8b-a727-d0136cd2ad40.png)

Reply:
perfect